lib/linguist/heuristics.rb

```diff
@@ -226,6 +226,16 @@ def call(data)
       end
     end
 
+    disambiguate ".gml" do |data|
+      if /^\s*(\<\?xml|xmlns)/i.match(data)
+        Language["XML"]
+      elsif /^\s*(graph\s?\[|node\s?\[)/i.match(data)
+        Language["Graph Modeling Language"]
+      else
+        Language["Game Maker Language"]
+      end
```

Comment:
I don't think we can be sure we've eliminated the two other candidate languages here. Are we even sure `/^\s*(graph\s?\[|node\s?\[)/i` will match all Graph Modeling Language files?

Reply:
As far as I could tell the Graph modeling Languages always have a ``graph [`` or ``node [`` element. 

http://www.fim.uni-passau.de/fileadmin/files/lehrstuhl/brandenburg/projekte/gml/gml-technical-report.pdf

I haven't found any docs, but I'm quite sure that it then will not be GML (``text [`` is invalid syntax), unless its in a comment ofcourse.

Maybe that would be a nicer test?

``([a-zA-Z]?\s?\[)`` being Graph Modeling Language?

Reply:
Here there is some more documentation. All files I've seen so far have that ``graph [`` and ``node [`` stuff.

http://www.fim.uni-passau.de/index.php?id=17297&L=1